discovery-infra/test_infra/cluster.py

```diff
@@ -0,0 +1,237 @@
+import logging
+
+from tests.conftest import env_variables
+from test_infra import consts, utils
+
+
+class Cluster:
+    
+    def __init__(self, api_client, cluster_name):
+        res = self._create(api_client, cluster_name)
+
+        self.id = res.id
+        self.api_client = api_client
+    
+    def _create(self, api_client, cluster_name):
+        return api_client.create_cluster(
+            cluster_name,
+            ssh_public_key=env_variables['ssh_public_key'],
+            openshift_version=env_variables['openshift_version'],
+            pull_secret=env_variables['pull_secret'],
+            base_dns_domain=env_variables['base_domain'],
+            vip_dhcp_allocation=env_variables['vip_dhcp_allocation']
+        )
+
+    def delete(self, api_client=None):
+        if api_client is None:
+            api_client = self.api_client
+
+        api_client.delete_cluster(self.id)
+
+
+    def generate_and_download_image(
+        self, 
+        api_client=None,
+        iso_download_path=env_variables['iso_download_path'],
+        ssh_key=env_variables['ssh_public_key']
+        ):
+        if api_client is None:
+            api_client = self.api_client
+        
+        api_client.generate_and_download_image(
+            cluster_id=self.id,
+            ssh_key=ssh_key,
+            image_path=iso_download_path
+        )
+
+    def wait_until_hosts_are_discovered(
+        self, api_client=None, nodes_count=env_variables['num_nodes']
+    ):
+        if api_client is None:
```

Comment:
if u need it for some reason, lets create decorator for this one, you can add this line to all functions.